### PR TITLE
Container

### LINKED ISSUE #83: Separate base Container and Device
There are a few LCLS specific things in `Device`. In reality we should have a base `Container` with the bare minimum globally useful keys, then have `Device` contain the more LCLS specific information i.e `stand`, `z`, `beamline`.

### PR DESCRIPTION
**Pull request by teddyrendahl**
## Description

For a while it has bothered me that base `happi.Device` makes a lot of assumptions about item you want to put in the database. I think location information should be required for physical devices but it is pretty unwieldy if you are just setting up a little simulation and want to fill a database with `ophyd.sim.SynAxis` objects.  

This PR creates a base `Container` object which has the bare minimum amount of `EntryInfo`. This is a good starting point for any external collaborators who want to use `happi` but don't plan on using any of our device types. One consequence of this is that we no longer enforce that `prefix` exists (see simulation) on `Container`. Therefore we need to retrain our database to use `name` as the unique enforceable key. This is still compatible with databases created with older versions of `happi` 

###### API Thoughts
In the docstrings I tried to replace all mentions of `Device` with `Container` but methods are still called `find_device` e.t.c. I like the new `Container` name, it creates a clear separation between the configuration information (`Container`) and the actual `ophyd.Device`. I think calling these the same thing is probably confusing for new folks.  

## Motivation and Context

Closes #83 

## How Has This Been Tested?

* Existing test suite passes, simply had to change a few places where we assumed what that `_id` tag was.
* Successfully loaded older happi database

## Where Has This Been Documented?

TBD

### CODE REVIEW

**Comment by ZLLentz** [comment]
Biggest issue for me now is that I get really confused over the device/container verbiage when reading the codebase, but maybe that should be addressed separately.

I'd ultimately like the word `device` to be reserved for an object that actually "does something", rather than one that just stores information.

So we'd have `Container` and `PCDSContainer`, for example, instead of calling the second one `Device`, with appropriately awful API breaks as needed.

**Comment by klauer** [comment]
I think the rename is important and definitely better to do sooner rather than later.  @ZLLentz and I came up with the following replacement class names:

* `Happi`
* `Descriptor`
* ~`Container`~
* ~`Device`~
* ~`Entry`~
* ~`Document`~
* ~`Row`~

We should brainstorm more...

**Comment by teddyrendahl** [comment]
Renamed to `HappiItem`. Added to docs. Ready for final review and merge.

File: happi/device.py [hunk 1]
```diff
@@ -382,3 +356,44 @@ def __repr__(self):
 
     def __eq__(self, other):
         return (self.prefix, self.name) == (other.prefix, other.name)
+
+
+class Device(HappiItem):
```
Comment (ZLLentz) [inline happi/device.py]: We should pick a reasonable rename alias for this when we transition away from using this name

File: happi/device.py [hunk 2]
```diff
@@ -382,3 +356,44 @@ def __repr__(self):
 
     def __eq__(self, other):
         return (self.prefix, self.name) == (other.prefix, other.name)
```
Comment (klauer) [inline happi/device.py]: Not part of the PR, but this jumps out to me as being insufficient.
It should be something like:
`(type(self) is type(other)) and all((self.* == other.*))`

Comment (teddyrendahl) [reply happi/device.py]: Good point. Let's make an issue

**Comment by teddyrendahl** [comment]
Had to rebase so I could get the `happi/cli.py` file that is causing the `flake8` exception on Travis 🤕

File: happi/device.py (hunk 2 above)
Comment (teddyrendahl) [reply happi/device.py]: actually adressing this now, `current  __eq__ depends on all devices having a prefix

**Comment by teddyrendahl** [comment]
A couple more issues. The old `__eq__` method required a `prefix` which is not on `HappiItem`. We now compare the full dump of the information that will be in the `happi` database. Not sure why we were not doing this before ...

The other thing is `happi` keeps a cache of the objects it loads. That cache was keyed on `prefix`... we now key on `name`. Finally, I noticed that if a device did not have metadata attached we would automatically return the cached object. This seems like the wrong behavior. If we can't guarantee the same device will be loaded we should side on reloading.

Worth another review @klauer @ZLLentz . Sorry for this dragging on.
